tests/test_examples.py

```diff
@@ -90,3 +110,14 @@ def test_cuda(example):
     filename, args = example[0], example[1:]
     filename = os.path.join(EXAMPLES_DIR, filename)
     check_call([sys.executable, filename] + args)
+
+
+@pytest.mark.skipif('CI' in os.environ, reason='slow test')
+@pytest.mark.xfail(reason='not jittable')
+@pytest.mark.parametrize('example', JIT_EXAMPLES)
+def test_jit(example):
+    logger.info('Running:\npython examples/{}'.format(example))
+    example = example.split()
+    filename, args = example[0], example[1:]
+    filename = os.path.join(EXAMPLES_DIR, filename)
+    check_call([sys.executable, filename] + args)
```

Comment:
if `CUDA_EXAMPLES` and `JIT_EXAMPLES` are subsets of `CPU_EXAMPLES` what do you think of writing the cuda and jit tests as:
```
@pytest.mark.parametrize('example', CPU_EXAMPLES)
def test_jit(example):
  try:
    ...
    filename, args = example[0], example[1:] + ' --jit'
    ...
  except CalledProcessError as e:
    if 'unrecognized arguments' in e.output:
    # if the examples does not support jit, skip it
      continue
```
or would that run too many needless examples?

Reply:
That seems like too much automation. I'd worry that as we add more logic we'd accidentally disable all tests and silently stop testing (this has happened a couple times already). Therefore I'm more comfortable hand-unrolling the examples lists.